stdlib/3/urllib/request.pyi

```diff
@@ -108,14 +108,14 @@ class HTTPPasswordMgr:
     def find_user_password(self, realm: str, authuri: str) -> Tuple[Optional[str], Optional[str]]: ...
 
 class HTTPPasswordMgrWithDefaultRealm(HTTPPasswordMgr):
-    def add_password(self, realm: str, uri: Union[str, Sequence[str]],
+    def add_password(self, realm: Optional[str], uri: Union[str, Sequence[str]],
                      user: str, passwd: str) -> None: ...
     def find_user_password(self, realm: str, authuri: str) -> Tuple[Optional[str], Optional[str]]: ...
```

Comment:
Seems like we should change the type of `realm` here as well

Reply:
[If I read it correctly](https://docs.python.org/3.8/library/urllib.request.html#urllib.request.HTTPPasswordMgr.find_user_password), `find_user_password` really expects a string as `realm`, but will search the `None`-realm if there is no direct match.

Reply:
Okay, yeah, I think you're right

Reply:
From the code it looks like None should work fine, and that's also what the docs for this class (https://docs.python.org/3.8/library/urllib.request.html#urllib.request.HTTPPasswordMgrWithDefaultRealm) suggest.

Reply:
Yup, None works fine for all the code (including of the superclass), which is why I made my original comment.

The issue is that [`find_user_password`](https://docs.python.org/3.8/library/urllib.request.html#urllib.request.HTTPPasswordMgr.find_user_password) docs make it sound like that's an implementation detail, since it's suggestive that the "given realm" is not None.

But given that it works, that evidently the docs are unclear, and that we err on the side of minimising false positives, we should probably just make realm Optional in find_user_password too.

Reply:
Note that those docs are for a base class that is *not* meant to accept None, while this stub is for a concrete child class that does.

Reply:
Good points all. I have made the realm optional in `find_user_password` too.